### PR TITLE
1198: Add variable_max_size to variables metadata for enhanced data insights

### PR DESCRIPTION
**Pull request by alexfurmenkov**

### CODE REVIEW

File: cdisc_rules_engine/dataset_builders/variables_metadata_dataset_builder.py
Comment (RamilCDISC) [inline cdisc_rules_engine/dataset_builders/variables_metadata_dataset_builder.py:1]: Could you please add testing for this new meta variable covering edge cases?

File: cdisc_rules_engine/dataset_builders/variables_metadata_dataset_builder.py [hunk 1]
```diff
@@ -12,7 +13,44 @@ def build(self):
...
-        return self.data_service.get_variables_metadata(
+        # Get basic variable metadata
+        variables_metadata = self.data_service.get_variables_metadata(
             dataset_name=self.dataset_path, datasets=self.datasets, drop_duplicates=True
         )
+
+        # Check if the rule requires variable_max_size
+        if (
+            self.rule
+            and self.rule.get("output_variables")
```
Comment (SFJohnson24) [inline cdisc_rules_engine/dataset_builders/variables_metadata_dataset_builder.py]: i think we want to change the conditions for this--a rule using variable_max_size may not put it in the output variables and cause issue downstream.
We would want to iterate through conditions, looking for variable_max_size--this shouldnt be too costly since most rules dont have too many conditions.  We will not know the shape of conditions though--could be any/all, nested so we will need recursion

Comment (alexfurmenkov) [reply cdisc_rules_engine/dataset_builders/variables_metadata_dataset_builder.py]: @SFJohnson24 Could you check the new logic?

**Comment by alexfurmenkov** [comment]
Hi, It's the rule and the negative dataset.
[1198.zip](https://github.com/user-attachments/files/26671254/1198.zip)

**Comment by RamilCDISC** [comment]
I have completed my review. But before approval we need to update the documentation to add this new variable. Once updated I will approve and final comment.
